e2e/sites.test.ts

```diff
@@ -8,81 +7,106 @@ import MainSidebar from './page-objects/main-sidebar';
+			const fullLocalPath = path.join( session.homePath, 'Studio', customFolderName );
+			await fs.mkdir( fullLocalPath, { recursive: true } );
+			env.E2E_OPEN_FOLDER_DIALOG = fullLocalPath;
+		}
+		await session.launch( env );
 
-		await expect( modal.createSiteButton ).toBeVisible();
-		await modal.createSiteButton.click();
+		const onboarding = new Onboarding( session.mainWindow );
 
-		await modal.siteNameInput.fill( name );
-		await modal.addSiteButton.click();
+		if ( customSiteName ) {
+			await onboarding.siteNameInput.fill( customSiteName );
+		}
+		await expect( onboarding.siteNameInput ).toHaveValue( /\S+/, { timeout: 5000 } );
+		const siteName = await onboarding.siteNameInput.inputValue();
 
-		const siteTitle = sidebar.getSiteNavButton( name );
-		await expect( siteTitle ).toHaveText( name );
+		if ( customFolderName ) {
+			await onboarding.selectLocalPathForTesting( customFolderName );
+		}
+		const localPath = await onboarding.localPathInput.inputValue();
 
-		const siteContent = new SiteContent( session.mainWindow, name );
-		await expect( siteContent.runningButton ).toBeAttached( { timeout: 120_000 } );
+		await expect( onboarding.heading ).toBeVisible();
+		await onboarding.continueButton.click();
 
-		return { sidebar, siteContent };
-	};
+		await closeWhatsNew();
 
-	test.beforeAll( async () => {
-		await session.launch();
+		const siteContent = new SiteContent( session.mainWindow, siteName );
+		await expect( siteContent.siteNameHeading ).toBeVisible( { timeout: 120_000 } );
 
-		// Complete onboarding before tests
-		const onboarding = new Onboarding( session.mainWindow );
-		await expect( onboarding.heading ).toBeVisible();
-		await onboarding.continueButton.click();
+		return {
+			siteName,
+			localPath,
+		};
+	}
 
+	async function closeWhatsNew() {
 		const whatsNewModal = new WhatsNewModal( session.mainWindow );
 		if ( await whatsNewModal.locator.isVisible( { timeout: 5000 } ) ) {
 			await whatsNewModal.closeButton.click();
 		}
+	}
 
-		const siteContent = new SiteContent( session.mainWindow, defaultSiteName );
-		await expect( siteContent.siteNameHeading ).toBeVisible( { timeout: 120_000 } );
-	} );
-
-	test.afterAll( async () => {
+	test.afterEach( async () => {
 		await session.cleanup();
 	} );
 
-	test( 'create a new site', async () => {
-		const { siteContent } = await createSite( siteName );
-
-		expect( await siteContent.siteNameHeading ).toHaveText( siteName );
-
-		expect(
-			await pathExists( path.join( session.homePath, 'Studio', siteName, 'wp-config.php' ) )
-		).toBe( true );
-
-		await siteContent.navigateToTab( 'Settings' );
-
-		expect( await siteContent.frontendButton ).toBeVisible();
-		const frontendUrl = await siteContent.frontendButton.textContent();
-		expect( frontendUrl ).not.toBeNull();
-		const response = await new Promise< http.IncomingMessage >( ( resolve, reject ) => {
-			http.get( `http://${ frontendUrl }`, resolve ).on( 'error', reject );
+	[
+		[ undefined, undefined ],
```

Comment:
What do you think about extracting this array to separate the constant to make it easier to read, and to document the intent via the constant name?